java/feathub-connectors/flink-connectors/flink-connector-redis/src/main/java/com/alibaba/feathub/flink/connectors/redis/sink/JedisMasterClient.java

```diff
@@ -38,25 +40,55 @@ public JedisMasterClient(String host, int port, String username, String password
             jedis.auth(password);
         }
         jedis.select(dbNum);
+        pipeline = null;
     }
 
     @Override
-    public long hset(byte[] key, Map<byte[], byte[]> hash) {
-        return this.jedis.hset(key, hash);
+    public void del(String key) {
+        if (pipeline == null) {
+            pipeline = jedis.pipelined();
+        }
+        pipeline.del(key);
     }
 
     @Override
-    public byte[] scriptLoad(byte[] script) {
-        return this.jedis.scriptLoad(script);
+    public void hmset(String key, Map<String, String> hash) {
+        if (pipeline == null) {
+            pipeline = jedis.pipelined();
+        }
+        pipeline.hmset(key, hash);
     }
 
     @Override
-    public Object evalsha(byte[] sha1, List<byte[]> keys, List<byte[]> args) {
-        return this.jedis.evalsha(sha1, keys, args);
+    public void rpush(String key, String... string) {
+        if (pipeline == null) {
+            pipeline = jedis.pipelined();
+        }
+        pipeline.rpush(key, string);
+    }
+
+    @Override
+    public void set(String key, String value) {
+        if (pipeline == null) {
+            pipeline = jedis.pipelined();
+        }
+        pipeline.set(key, value);
+    }
+
+    @Override
+    public void flush() {
+        if (pipeline != null) {
+            pipeline.sync();
```

Comment:
The JavaDoc of `redis.clients.jedis.Pipeline#sync` shows the following contents.
```java
  /**
   * Synchronize pipeline by reading all responses. This operation close the pipeline. In order to
   * get return values from pipelined commands, capture the different Response&lt;?&gt; of the
   * commands you execute.
   */
```
As this operation closes the pipeline, I supposed that we need to re-instantiate the pipeline then.

However, I found that Feathub is still able to write out content to Redis using the same pipeline after it is synced and closed, so it might be better to reuse one pipeline for writing multiple rows of features.

Reply:
Sounds good.